sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -188,7 +188,7 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
   private def verifySchema(schema: StructType): Unit = {
     schema.foreach { field =>
       field.dataType match {
-        case _: ArrayType | _: MapType | _: StructType =>
+        case _: ArrayType | _: MapType | _: StructType | NullType | CalendarIntervalType =>
```

Comment:
It seems no tests about UDT for now but it uses `UserDefinedType.sqlType` for writing and throws an exception for reading. I can make them consistent here. Also, let me try to Whitelist them. Thank you so much.